modules/reranker-voyageai/module.go

```diff
@@ -0,0 +1,91 @@
+//                           _       _
+// __      _____  __ ___   ___  __ _| |_ ___
+// \ \ /\ / / _ \/ _` \ \ / / |/ _` | __/ _ \
+//  \ V  V /  __/ (_| |\ V /| | (_| | ||  __/
+//   \_/\_/ \___|\__,_| \_/ |_|\__,_|\__\___|
+//
+//  Copyright © 2016 - 2024 Weaviate B.V. All rights reserved.
+//
+//  CONTACT: hello@weaviate.io
+//
+
+package modrerankervoyageai
+
+import (
+	"context"
+	"net/http"
+	"os"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"github.com/weaviate/weaviate/entities/modulecapabilities"
+	"github.com/weaviate/weaviate/entities/moduletools"
+	"github.com/weaviate/weaviate/modules/reranker-voyageai/clients"
+	rerankeradditional "github.com/weaviate/weaviate/usecases/modulecomponents/additional"
+	"github.com/weaviate/weaviate/usecases/modulecomponents/ent"
+)
+
+const Name = "reranker-voyageai"
+
+func New() *ReRankerVoyageAIModule {
+	return &ReRankerVoyageAIModule{}
+}
+
+type ReRankerVoyageAIModule struct {
+	reranker                     ReRankerVoyageAIClient
+	additionalPropertiesProvider modulecapabilities.AdditionalProperties
+}
+
+type ReRankerVoyageAIClient interface {
+	Rank(ctx context.Context, query string, documents []string, cfg moduletools.ClassConfig) (*ent.RankResult, error)
+	MetaInfo() (map[string]interface{}, error)
+}
+
+func (m *ReRankerVoyageAIModule) Name() string {
+	return Name
+}
+
+func (m *ReRankerVoyageAIModule) Type() modulecapabilities.ModuleType {
+	return modulecapabilities.Text2TextReranker
+}
+
+func (m *ReRankerVoyageAIModule) Init(ctx context.Context,
+	params moduletools.ModuleInitParams,
+) error {
+	if err := m.initAdditional(ctx, params.GetConfig().ModuleHttpClientTimeout, params.GetLogger()); err != nil {
+		return errors.Wrap(err, "init cross encoder")
+	}
+
+	return nil
+}
+
+func (m *ReRankerVoyageAIModule) initAdditional(ctx context.Context, timeout time.Duration,
+	logger logrus.FieldLogger,
+) error {
+	apiKey := os.Getenv("VOYAGE_API_KEY")
```

Comment:
please use the same env variable as in vectorizer module which is: `VOYAGEAI_APIKEY`